File: notags/GroupthePeopleGiventheGroupSizeTheyBelongTo.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

public class GroupthePeopleGiventheGroupSizeTheyBelongTo {

    public List<List<Integer>> groupThePeople(int[] groupSizes) {
        int len = groupSizes.length;
        Map<Integer, ArrayList> map = new HashMap<Integer, ArrayList>();

        for (int i = 0; i < groupSizes.length; i++) {
            if (map.get(groupSizes[i]) == null) {
                ArrayList<Integer> arrayList = new ArrayList<>();
                arrayList.add(i);
                map.put(groupSizes[i], arrayList);
            }

            else
                map.get(groupSizes[i]).add(i);
        }
        List<List<Integer>> res;
        for(int size:map.keySet()){
            for(int i=0;i<map.get(size).size();i+=size)

        }

    }

    public static void main(String[] args) {

    }
}

File: notags/GroupthePeopleGiventheGroupSizeTheyBelongTo.py


class Solution:
    def groupThePeople(self, groupSizes):
        A = {}
        for i,size in enumerate(groupSizes):
            if size not in A:
                A[size] = []
            A[size].append(i)

        res = []
        for size in A:
            for i in range(0,A[size],size):
                res.append(A[size][i:i+size])
        return res

if __name__ == "__main__":
    a = Solution()
    print(a.groupThePeople([3,3,3,3,3,1,3]))
    print("h")